src/App.vue

```diff
@@ -425,39 +425,42 @@ export default class App extends Mixins(AuthApiMixin, CommonMixin, DateMixin, Fi
     this.$root.$off('go-to-dashboard')
   }
 
-  /** Called when $route property changes. Used to init app. */
-  @Watch('$route', { immediate: true })
+  /** Called when $route property changes. */
+  @Watch('$route', { immediate: false })
   private async onRouteChanged (): Promise<void> {
     // init only if we are not on signin or signout route
     if (!this.isRouteName(RouteNames.SIGN_IN) && !this.isRouteName(RouteNames.SIGN_OUT)) {
       // store current filing type
       const filingType = this.$route.matched[0]?.meta.filingType
       filingType && this.setFilingType(filingType)
 
-      // get and store business ID
+      // get and store Business ID
       const businessId = sessionStorage.getItem('BUSINESS_ID')
       this.setBusinessId(businessId)
 
       // load account information
       this.loadAccountInformation()
 
       // initialize app
-      await this.initApp(true)
+      await this.fetchData(true)
     }
   }
 
   //
   // http://localhost:8080/basePath/BC0870829/alteration
   //
 
-  /** Initializes application. Also called for retry. */
-  private async initApp (routeChanged: boolean = false): Promise<void> {
+  /** Fetches app data. Also called for retry. */
+  private async fetchData (routeChanged: boolean = false): Promise<void> {
     // only fetch data on first route change
     if (routeChanged && this.haveData) return
 
     // reset errors in case of retry
     this.resetFlags()
 
+    // set current date from "real time" date from server
+    this.setCurrentDate(this.dateToYyyyMmDd(this.getCurrentJsDate))
```

Comment:
I just moved this up (was at bottom of this function before) in case any other code in here depends on it. This is also now consistent with Create UI.